src/aws/ec2Instances.ts

```diff
@@ -25,3 +38,29 @@ export function getSpecificInstance(instanceIds: string[], instanceFilter: Insta
     )
 }
 
+export async function getInstancesByTag(tagKey: string): Promise<Instance[]> {
+    console.log(`Finding EC2 instances that have tag ${tagKey}`);
+
+    async function _getInstancesByTag(acc: Instance[] = [], nextToken?: string): Promise<Instance[]> {
+        const params = {
+            MaxResults: 1000,
+            NextToken: nextToken,
+            Filters: [
+                { "Name": "tag-key", Values: [tagKey] }
+            ]
+        };
+
+        const response = await awsEc2.describeInstances(params).promise();
+        
+        const instances = buildInstances(response);
+        const allInstances = acc.concat(instances);
+
+        if(response.NextToken) {
+            return await _getInstancesByTag(allInstances, response.NextToken);
```

Comment:
Recursion being useful, yay! 😄

Reply:
Not just any recursion, asynchronous recursion ↗️ 🚀 🌕